cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -794,36 +814,36 @@ private KafkaRebalanceStatus buildRebalanceStatus(String sessionID, KafkaRebalan
         return apiClient.rebalance(host, CruiseControl.REST_API_PORT, rebalanceOptionsBuilder.build(), userTaskID)
                 .map(response -> {
                     if (dryrun) {
-                        if (response.thereIsNotEnoughDataForProposal()) {
-                            // If there are not enough data for a rebalance, it's an actual error at Cruise Control level
-                            // and we need to re-request the proposal at a later stage so we move to the PendingProposal State.
+                        if (!response.isSufficientDataForProposal()) {
```

Comment:
I think that @tomncooper was using the logic with `thereIsNotEnough....` because the Cruise Control has something similar about the window in which you need samples for data. I would leave that logic tbh to have the same kind of mapping with Cruise Control logic. Starting to use the opposite logic in the code and then negate it ... it's quite confusing. Just my 2 cents.

Reply:
The "NotEnoughData" flag is based on the "NotEnoughValidWindows" error on the CC side. It is an exceptional state (hopefully) so we went with a flag that has to be actively set to true when the error occurs. I am not clear what the advantage is to swapping this around?